src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -677,17 +677,24 @@ private enum ReceiverCaptureMode
                 }
                 else
                 {
-                    refKind = rewrittenReceiver.GetRefKind();
-
-                    if (refKind == RefKind.None &&
-                        !rewrittenReceiver.Type.IsReferenceType &&
-                        Binder.HasHome(rewrittenReceiver,
-                                       Binder.AddressKind.Constrained,
-                                       _factory.CurrentFunction,
-                                       peVerifyCompatEnabled: false,
-                                       stackLocalsOpt: null))
+                    if (rewrittenReceiver.Type.IsReferenceType)
```

Comment:
Copying comment from extensions/receiver PR in case it is useful:
```
            // If we need to capture a receiver (for compound assignment, null-coalescing assignment, extension receiver adjustment, interpolation `this` capture, ...)
            // we need to make sure the capturing temp preserves the semantics of the receiver.
            //
            // For instance:
            //   Struct s = ...;
            //   s.Method(s.field = 42)
            // must be rewritten as:
            //   ref Struct temp = ref s;
            //   temp.Method(s.field = 42)
            // so that Method can have and observe the proper side-effects on the struct.
            //
            // Similarly:
            //   Class c = ...;
            //   ref Class c2 = ref c;
            //   Class otherC = null;
            //   c2.Method(c2 = ref otherC)
            // must be rewritten as:
            //   Class temp = c2; // dereferences to c
            //   temp.Method(c2 = ref otherC)
```